pyspecProcScripts/calc_error.py

```diff
@@ -0,0 +1,102 @@
+from .simple_functions import select_pathway
+import numpy as np
+
+
+# {{{ Functions to calculate var and mean of nan-masked data
+#     Pulled from time_domain_noise.py example
+def _masked_mean_multi(x, axis=None):
+    "Calculate the mean of nan-masked data on a 1D axis"
+    assert axis is not None
+
+    def masked_mean(x):
+        return np.mean(x[np.isfinite(x)])
+
+    return np.apply_along_axis(masked_mean, axis, x)
+
+
+def _masked_var_multi(x, var_has_imag=False, axis=None):
+    """Calculates the variance along a 1D axis.
+    If the data is complex you must assign var_has_imag as true
+    so that the calculated variance is divided by 2.
+    By default it calculates the variance of the real of the data"""
```

Comment:
why can't it just figure  out if it has an imaginary part?

Reply:
throws a ValueError now if this passes